### ISSUE TITLE
When I use an Hosted BackgroundService (which takes in an MEL logger) it's logs do not get output into serilog.

### ISSUE THREAD

**Issue by AraHaan** [serilog/serilog#1691]
**Description**
I have an problem with my Discord Bot not logging from BackgroundService's that it uses but does not have in it's project (the project for those services do not reference any of the serilog packages directly). As such they oddly do not get logged into Serilog or any of my sinks despite configuring them to do so.

**Reproduction**

Create console application that uses the AspNetCore framework but also sets these:
```xml
    <NoDefaultLaunchSettingsFile>true</NoDefaultLaunchSettingsFile>
    <!-- When the program is published web.config is generated which is not intended at all. -->
    <IsWebConfigTransformDisabled>true</IsWebConfigTransformDisabled>
```

Code:
```cs
// other usings in global usings in project settings.
using Serilog;

AppDomain.CurrentDomain.UnhandledException += (s, e) =>
{
    // log the exception to a file.
    SlashUpdateService.UnhandledException((Exception)e.ExceptionObject);
    Environment.Exit(1);
};
var executingAssembly = Assembly.GetExecutingAssembly();
Console.Title = string.Format(
    Resources.BotTitle!,
    executingAssembly.GetName().Name,
    executingAssembly.GetName().Version!.ToString(3));
DiscordBotSingleton.CleanAllLogs();
// yes I hijack the AspNetCore framework here to run a Discord Bot instead which helps reduce about 19 assemblies from packages (but I want more to be reduced).
await Host.CreateDefaultBuilder()
    .UseContentRoot(Path.GetDirectoryName(executingAssembly.Location)!)
    .UseSerilog((hostingContext, services, loggerConfiguration) => loggerConfiguration
        .ReadFrom.Configuration(hostingContext.Configuration))
    .ConfigureServices((hostContext, collection) =>
        // The background services gets registered here.
        DiscordBotSingleton.ConfigureServices(
            hostContext.Configuration.GetSection("connectionString").Value!,
            collection))
    .RunConsoleAsync();
```

**Expected behavior**
For the logs to be in the serilog outputs.

**Relevant package, tooling and runtime versions**
What Serilog version are you using, on what platform?

used runtime versions: 6.0.6 servicing releases of the .NET shared frameworks.

Serilog versions:
  - Serilog.Expressions 3.4.0
  - Serilog.Extensions.Hosting 5.0.0-dev-00095
  - Serilog.Settings.Configuration 3.3.1-dev-00337
  - Serilog.Sinks.File 5.0.1-dev-00947

**Additional context**
N/A (I think).

**Comment by bartelink** [comment]
As this is unlikely to be a bug in the core Serilog lib, I'd suggest asking on Stack Overflow with a `serilog` tag first to get more eyes on it - the chances are someone has tried something similar before. (Issues on this repo consume valuable time of the maintainers, so seeing whether there are workarounds or known integration issues is an important first step in order to conserve that). (Also, `UseSerilog` and the config reading logic does not live in this repo)

In general I'd recommend ruling out config file parsing issues by putting in more direct configuration in code and see if that helps.

And an aside: one thing to look out for when doing `System.Exit` is that doing a `Log.CloseAndFlush` is required if you want to guarantee output.

**Comment by AraHaan** [comment]
I actually figured out the issue, the issue happened when I made an shared framework for using "serilog" because dotnet store does not work in .NET 6+ and so I had no other option to share the serilog binaries with multiple projects other than using the dotnet/arcade sharedframework sdk and so then it builds serilog into a nice little installer (with it crossgen2'd to make it faster with ready to run).

What caused it to happen? turns out that when doing such a thing the Serilog.Settings.Configuration could not find the sinks and so I had to manually specify them in the json file with ``    "Using": [ "Serilog.Sinks.File", "Serilog.Expressions" ],``. After that the logs started to be written.

If you guys are willing to, I am willing to contribute the installer option to serilog so that way others who want to use serilog (not as a package but as an installable framework) can do so and save disk space on  installing the serilog packages:

Note: I might have accidentally left out some maintained packages from serilog as I was not sure which all was obsolete or outdated and unmaintained (is archived) and so a select few packages might be present in that shared framework.
